worlds/oot_soh/ItemPool.py

```diff
@@ -250,6 +242,19 @@ def create_item_pool(world: "SohWorld") -> None:
     if world.options.shuffle_dungeon_rewards == "dungeons":
         filler_item_count -= 9
 
+    # Triforce pieces
+    if world.options.triforce_hunt:
+        total_triforce_pieces = min(
+            filler_item_count, world.options.triforce_hunt_pieces_total.value)
+
+        for _ in range(total_triforce_pieces):
+            world.item_pool += [world.create_item(Items.TRIFORCE_PIECE)]
+
+        filler_item_count -= total_triforce_pieces
+
+        if world.options.triforce_hunt_pieces_total.value > total_triforce_pieces:
+            world.options.triforce_hunt_pieces_total.value = total_triforce_pieces
+
```

Comment:
I'd rather move the required triforce piece calculation to here and save it to world.triforce_pieces_required (not the option, a separate variable in world). Then pass those into the logic rule and slot data instead of what you currently have in `self.options.triforce_hunt_pieces_required`. 

Also, I'm pretty sure the way you're calculating the rule for the actual triforce hunt can lead to the required pieces evaluating to 0 because python floors partials when they're used as ints. Probably want to special case this if it's 0, set it to 1 instead.